src/dials/util/table_as_hdf5_file.py

```diff
@@ -0,0 +1,271 @@
+
+        for key, data in table.cols():
+            if ignore and key in ignore:
+                continue
+            if isinstance(data, flex.shoebox):
+                ReflectionListEncoder.encode_shoebox(group, data, key)
+            elif isinstance(data, flex.int6):
+                s = data.size()
+                this_data = flumpy.to_numpy(data.as_int()).reshape(s, 6)
+                group.create_dataset(
+                    key, data=this_data, shape=this_data.shape, dtype=this_data.dtype
+                )
+            elif isinstance(data, flex.std_string):
+                this_data = [s.encode("utf-8") for s in data]
+                group.create_dataset(key, data=this_data, shape=(len(this_data),))
+            else:
+                this_data = flumpy.to_numpy(data)
+                group.create_dataset(
+                    key, data=this_data, shape=this_data.shape, dtype=this_data.dtype
+                )
+
+    @staticmethod
+    def encode_shoebox(group: h5py.Group, data: flex.shoebox, key: str):
+        """Encode a column of shoeboxes."""
+        sbdata, bg, mask = data.get_shoebox_data_arrays()
+        panel = data.panels()
+        bbox = data.bounding_boxes()
+        data = flumpy.to_numpy(sbdata)
+        lz4 = hdf5plugin.LZ4()
+        sbox_group = group.create_group(key)
+        sbox_group.create_dataset(
+            "shoebox_data",
+            data=data,
+            shape=data.shape,
+            dtype=data.dtype,
+            compression=lz4,
+        )
+        data = flumpy.to_numpy(bg)
+        sbox_group.create_dataset(
+            "shoebox_background",
+            data=data,
+            shape=data.shape,
+            dtype=data.dtype,
+            compression=lz4,
+        )
+        data = flumpy.to_numpy(mask)
+        sbox_group.create_dataset(
+            "shoebox_mask",
+            data=data,
+            shape=data.shape,
+            dtype=data.dtype,
+            compression=lz4,
+        )
+        s = bbox.size()
+        this_data = flumpy.to_numpy(bbox.as_int()).reshape(s, 6)
+        sbox_group.create_dataset(
+            "bbox",
+            data=this_data,
+            shape=this_data.shape,
+            dtype=this_data.dtype,
+            compression=lz4,
+        )
+        data = flumpy.to_numpy(panel)
+        sbox_group.create_dataset(
+            "panel", data=data, shape=data.shape, dtype=data.dtype, compression=lz4
+        )
+
+
+class ReflectionListDecoder(object):
+    """Decoder for the reflection data."""
+
+    @staticmethod
+    def decode(handle: h5py.File) -> List[flex.reflection_table]:
+        """Decode the data to a list of reflection tables."""
+
+        # Get the group containing the reflection data
+        g = handle["entry/data_processing"]
+
+        # Create the list of reflection tables
+        tables = []
+        for dataset in g.values():
+            if "num_reflections" not in dataset.attrs:
```

Comment:
What if the `num_reflection` differs from the dataset shape? Can we not just get this from the shape of the HDF5 data sets?

Reply:
Agreed, removed this attribute